static-site/src/components/ListResources/types.ts

```diff
@@ -44,13 +44,20 @@ export interface UpdateCadence {
   description: string
 }
 
-// See coreShowcase in static-site/content/resource-listing.yaml
 export interface Card {
```

Comment:
The splitting of `Cards` to `FilterCard` and `UrlCard` was motivated by type assertions which I've since realized are [no good](https://github.com/nextstrain/nextstrain.org/pull/874#discussion_r1623070378) and resulted in poor error handling. I decided to revert the split and instead make `filters` and `url` optional: 70b94a67

Reply:
[Not a request for changes]

My reading of Ivan's post is that it's focused on avoiding casts. That seems good. But I thought using separate types for separate functionalities (and a filter card vs a url card seem like perfect candidates for such separate functionalities) was a key point in using types. 70b94a6 adds a bunch of runtime error checking (`if (!card.filters) { throw ... `) which I thought static type checking was supposed to avoid. I haven't used TS enough to make a call here, but I'm surprised at how much runtime checking we're still having to do.

Reply:
That's a good point. I've pushed another commit which re-introduces `FilterCard` and removes the need for extra runtime checks: 694f32fd

The key here was figuring out how to allow any type that extends `Card` to be used in `Showcase` (not hardcoding to `FilterCard[] | UrlCard[]`). I did this by using [generic constraints](https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-constraints):

https://github.com/nextstrain/nextstrain.org/blob/e6fa8aa098ca085d170ab2084788e1324d7f0e9f/static-site/src/components/Showcase/index.tsx#L14-L19